### DISCUSSION TITLE
Cannot convert strings to category/factor/enum

### DISCUSSION THREAD

**Discussion by david-cortes** [duckdb/duckdb#9509] [Ideas]
I want to have a query result with categorical columns being encoded as pandas 'Category' or R 'factor' type for better memory efficiency. Trying to cast a string (python 'object' or R 'character' types) in a query does not produce the desired result:
```python
import pandas as pd
df = pd.DataFrame({"col1" : ["a","b","c"]})
import duckdb
duckdb.query("select col1::enum from df").to_df()
```
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-2-81ef41169f6a> in <module>
      2 df = pd.DataFrame({"col1" : ["a","b","c"]})
      3 import duckdb
----> 4 duckdb.query("select col1::enum from df").to_df()

RuntimeError: Catalog Error: Type with name enum does not exist!
Did you mean "null"?
```

I would have to manually create a type and assign it there, which is extremely inconvenient and pollutes my duckdb namespace:
```python
import duckdb
duckdb.query("create type my_col_type as enum('a', 'b', 'c');")
import pandas as pd
df = pd.DataFrame({"col1" : ["a","b","c"]})
duckdb.query("select col1::my_col_type from df").to_df()
```

**Comment by Alex-Monahan** [comment]
Today, this behavior is expected, but it is on the roadmap to enable the syntax you are interested in!

**Comment by david-cortes** [comment]
Still relevant.

**Reply by Tishj** [reply]
You can cast to enum, no need to create a type schema entry

It requires the full definition, not just `enum`

**Comment by david-cortes** [comment]
> You can cast to enum, no need to create a type schema entry
> 
> It requires the full definition, not just `enum`

Still, it requires listing the entries, it doesn't allow something like `select col::enum` the same way for example polars would allow `col.cast(pl.Categorical)`. So it cannot be used for casting string columns to categorical which are the result of a long query where the final possible categories are not known apriori.
